public/javascripts/items.js

```diff
@@ -7,28 +7,105 @@ const fetchParams = {
   credentials: 'include'
 }
 
-const params = completed =>
-  Object.assign( {}, fetchParams, { body: JSON.stringify({ completed }) })
+const params = data =>
+  Object.assign( {}, FETCH_PARAMS, { body: JSON.stringify( data ) } )
 
-$(document).ready( () => {
-  $( '.title' ).click( event => {
-    const element = $( event.target )
-    const id = element.data( 'id' )
-    const completed = ! element.data( 'completed' )
+const RETURN_KEY = 13
+
+const checkJsonForSuccessField = json => {
+  if( json.success ) {
+    Promise.resolve( json )
+  } else {
+    Promise.reject( json.message )
+  }
+}
+
+const titleEdited = event => {
+  const element = $( event.target )
+  const id = element.data( 'id' )
+
+  if( event.charCode === RETURN_KEY ) {
+    let updatedTitle = element[0].value
+    fetch( `/items/${id}/edit`, params( { title: updatedTitle } ) )
+    .then( result => result.json() )
+    .then ( checkJsonForSuccessField )
+    .then( json => {
+      let span = $( `.title[data-id=${id}] > span`)
+      $( span[0] ).html(updatedTitle)
+      element.addClass( 'hidden')
+      $( span ).removeClass( 'hidden' )
+    })
+    .catch( error => {
+      console.log('error:', error)
+    })
+  }
+}
+
+const titleClicked = event => {
+  const element = $( event.target )
+  const id = element.data( 'id' )
+  $( element[0] ).addClass( 'hidden' )
+  let input = $( `.title[data-id=${id}] > input`)
+  $( input[0]).removeClass( 'hidden' )
+}
 
-    fetch( `/items/${id}/completed`, params( completed ))
+const descriptionEdited = event => {
+  const element = $( event.target )
+  const id = element.data( 'id' )
+
+  if( event.charCode === RETURN_KEY ) {
+    let updatedDescription = element[0].value
+    fetch( `/items/${id}/edit`, params( { description: updatedDescription } ) )
       .then( result => result.json() )
+      .then( checkJsonForSuccessField )
       .then( json => {
-        if( json.success ) {
-          element.data( 'completed', completed )
-
-          const parent = $( `.item[data-id=${id}]` )
-          if( completed ) {
-            parent.addClass( 'completed' )
-          } else {
-            parent.removeClass( 'completed' )
-          }
-        }
+        let span = $(`.description[data-id=${id}] > span`)
+        $( span[0] ).html( updatedDescription )
+        element.addClass( 'hidden')
+        $( span ).removeClass( 'hidden' )
```

Comment:
I just didn't see the similarity, mostly because I look at the code line by line lol and lose track of the global context sometimes. If one function can be created to elegantly handle all four cases, then yes, I think we should consolidate the logic.

You can change the above function to accommodate the span by adding another parameter:
```js
const toggle = (className, id, text, parent, childTag) => {
  const element = $( `.${className}[data-id=${id}] >${childTag}` )
  // etc.
}
```